### ISSUE TITLE
Provide a security.txt

### ISSUE THREAD

**Issue by ziegenberg** [opencast/opencast#6324]
I propose implementing RFC 9116 and providing a security.txt. We would add one file per Opencast install. We could have a look at how Moodle does it. They point to a central security.txt on their servers. See https://tuwel.tuwien.ac.at/security.txt for example.

It would need a regular update with every release to set a new `Expires` date.

```txt
# Moodle HQ security submission form
Contact: https://moodle.org/security/report/

# Moodle LMS security publications and acknowledgements page
Acknowledgments: https://moodle.org/security/

# Submission preferred languages
Preferred-Languages: en

# Moodle HQ canonical security.txt file
Canonical: https://moodle.org/.well-known/security.txt

# Moodle LMS Security Procedures document, including Responsible Disclosure Policy
Policy: https://moodledev.io/general/development/process/security

# Expiry date of this document
# This information is considered current and up to date until 3 weeks after the next major Moodle LMS release
Expires: 2025-05-05T01:00:00.000Z
```

**Comment by mtneug** [comment]
Wait so the idea is to have one `security.txt` that is the same for every Opencast instance? Why? The Opencast project does solve issues in the Opencast code base but we don't fix issues with specific instances. So I would expect to have instance-specific `security.txt` responses.

**Comment by gregorydlogan** [comment]
Yep, not sure I'm understanding here.  So this would be like the Terms of Use thing for the admin UI where the adopter needs to set this file up and remember to keep it updated?

**Comment by ziegenberg** [comment]
We could provide a default `security.txt` file, which we update with every release. This can be automated and templated in GH actions. We can set the expiration date to a time frame in which we expect the next but one version. Each adopter can edit and adjust the file to meet his own needs. But if they do not edit it, there is at least a security.txt, and we can get notified if someone finds a vulnerability. I'm sure most incidents will be code-based and not configuration-based.

Every Moodle installation has a default `security.txt` pointing to the Moodle HQ.

**Comment by mtneug** [comment]
I feel very uneasy about this default `security.txt`. To me, it feels like we are saying the Opencast project takes responsibility for the security of every Opencast installation in a direct sense. And on the other side as an admin, I don't want people to always tell the Opencast project about security issues if they might be related to my configuration.
